src/atomWithQuery.ts

```diff
@@ -0,0 +1,57 @@
+import type { AnyVariables, Client, Operation } from '@urql/core'
+import { createRequest } from '@urql/core'
+import type { Getter } from 'jotai/vanilla'
+import { clientAtom } from './clientAtom'
+import { createAtoms } from './common'
+import { AtomWithQuery, AtomWithQueryOptions } from './types'
+
+export function atomWithQuery<
+  Data = unknown,
+  Variables extends AnyVariables = AnyVariables
+>(
+  options: AtomWithQueryOptions<Data, Variables>
+): AtomWithQuery<Data, Variables> {
+  const {
+    query,
+    getVariables = () => ({} as Variables),
+    getContext,
+    getClient = (get: Getter) => get(clientAtom),
+    getPause: getPause = () => false,
+  } = options
+  const cache = new WeakMap<Client, Operation>()
```

Comment:
This cache is shared across Jotai providers, unless it's your intention, we should wrap it in an atom.
```js
const cacheAtom = atom(() => new WeakMap<Client, Operation>())
```
which creates a new weak map for each provider (store).

Reply:
It gets created when atomWithQuery gets called and then it gets closured into the callbacks defined below. Every time atomWithQuery will be called, this local WeakMap will be created. I don't see how it will be shared across other atoms and providers. If that would be an atom, then it may be shared in case it's instantiated outside of the AtomWithQuery function. But in this case I don't see how it gets into the global scope of things.

Reply:
I can try creating a contrived example. This also affects the following comments.

Reply:
Removed this